File: rider-app/src/app/index/DriverMatch.ts
import { DriverLocation } from "./DriverLocation";
import { Request } from "./Request";

export interface DriverMatch {
    stage: DriverMatchStage,
    request: Request,
    driverLocation: DriverLocation
}

export enum DriverMatchStage {
    MATCH,
    ACCEPT,
    REJECT,
    DRIVER_NO_FOUND
}

File: driver-backend/src/main/java/com/irembo/ride/driver/rider/DriverMatch.java
package com.irembo.ride.driver.rider;

import com.irembo.ride.driver.driverlocation.DriverLocation;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class DriverMatch {

    private DriverMatchStage stage;
    private Request request;
    private DriverLocation driverLocation;
}
